src/ucp/proto/proto_common.inl

```diff
@@ -371,4 +368,79 @@ ucp_proto_request_pack_rkey(ucp_request_t *req, ucp_md_map_t md_map,
     return packed_rkey_size;
 }
 
+static UCS_F_ALWAYS_INLINE size_t
+ucp_proto_request_pack_iov_rkeys(ucp_request_t *req, ucp_md_map_t md_map,
+                                 uint64_t distance_dev_map,
+                                 const ucs_sys_dev_distance_t *dev_distance,
+                                 void *rkey_buffer)
+{
+    const ucp_datatype_iter_t *dt_iter = &req->send.state.dt_iter;
+    ucp_mem_h *memh                    = dt_iter->type.iov.memh;
+    void *p                            = rkey_buffer;
+    size_t iov_count, iov_index;
+    size_t packed_count, *ppacked_count, *ppacked_rkey_size;
+    ssize_t packed_rkey_size;
+    size_t total_size;
+
+    ucs_assert(dt_iter->dt_class == UCP_DATATYPE_IOV);
+    if (md_map == 0 || dt_iter->length == 0) {
+        return 0;
+    }
+
+    /* 
+     * packed IOV rkeys format:
+     * | count(32/64)| address(64)| length(32/64)| rkey size(32/64)| rkey buffer|
+     *               | address(64)| length(32/64)| rkey size(32/64)| rkey buffer|
+     *               | address(64)| length(32/64)| rkey size(32/64)| rkey buffer|
+     */
+    ppacked_count = (size_t*)p;
+    p             = UCS_PTR_BYTE_OFFSET(p, sizeof(size_t));
+    iov_count     = ucp_datatype_iter_iov_count(dt_iter);
+    packed_count  = iov_count;
+    total_size    = 0;
+    for (iov_index = 0; iov_index != iov_count; ++iov_index) {
+        if (dt_iter->type.iov.iov[iov_index].length == 0) {
+            ucs_info("iov[%lu] block size is 0", iov_index);
```

Comment:
pls use `ucs_data` as it is supposed to be printed for every msg
`ucs_info` is for important notifications printed quite rarely

Reply:
done